doc/book/cookbook/using-a-base-path.md

```diff
@@ -0,0 +1,176 @@
+
+```bash
+$ cp config/autoload/los-basepath.global.php config/autoload/los-basepath.local.php
+```
+
+Then edit one or both, to change the `los_basepath` settings:
+
+```php
+return [
+    'los_basepath' => '<base path here>',
+    /* ... */
+];
+```
+
+The base path should be the portion of the web root leading up to the
+`index.php` of your application. In the above example, this would be
+`/expressive`.
+
+### mtymek/blast-base-url
+
+To use `mtymek/blast-base-url`, install it via Composer, and register some
+configuration.
+
+To install it:
+
+```bash
+$ composer require "mtymek/blast-base-url:dev-master@dev"
+```
+
+To configure it, update the file `config/autoload/middleware-pipeline.global.php`,
+with the following contents:
+
+```php
+return [
+    'dependencies' => [
+        'factories' => [
+            Blast\BaseUrl\BaseUrlMiddleware::class => Blast\BaseUrl\BaseUrlMiddlewareFactory::class,
+            /* ... */
+        ],
+        /* ... */
+    ],
+    'middleware_pipeline' => [
+        'pre_routing' => [
+            [ 'middleware' => [ BaseUrlMiddleware::class ] ],
+            /* ... */
+        ],
+        /* ... */
+    ],
+];
+```
+
+At this point, the middleware will take care of the rewriting for you. No
+configuration is necessary, as it does auto-detection of the base path based on
+the request URI and the operating system path to the application.
+
+The primary advantage of `mtymek/blast-base-url` is in its additional features:
+
+- it provides an override of `Zend\Expressive\Helper\UrlHelper` that is aware of
+  the base path, allowing you to create route-based URLs relative to the base
+  path.
+- it provides a new helper, `Blast\BaseUrl\BasePathHelper`, which allows you to
+  create URLs relative to the base path; this is particularly useful for assets.
+
+To enable these features, we'll add some configuration to
+`config/autoload/dependencies.global.php` file:
+
+```php
+return [
+    'dependencies' => [
+        'invokables' => [
+            Blast\BaseUrl\BasePathHelper::class => Blast\BaseUrl\BasePathHelper::class,            
+            /* ... */
+        ],
+        'factories' => [
+            Blast\BaseUrl\UrlHelper::class => Blast\BaseUrl\UrlHelperFactory::class,            
+            /* ... */
+        ],
+        'aliases' => [
+            // alias default UrlHelper with Blast\BaseUrl alternative
+            Zend\Expressive\Helper\UrlHelper::class => Blast\BaseUrl\UrlHelper::class,
+            /* ... */
+        ],
+    ],
+];
+```
+
+Finally, if you're using zend-view, you can register a new "basePath" helper in
+your `config/autoload/templates.global.php`:
+
+```
+return [
+    /* ... */
+    'view_helpers' => [
+        'factories' => [
+            'basePath' => Blast\Baseurl\BasePathViewHelperFactory::class,
+            /* ... */
+        ],
+        /* ... */
+    ],
+];
+```
+
+Usage of the `BasePath` helper is as follows:
+
+```php
+echo $basePathHelper('/icons/favicon.ico');
```

Comment:
This one won't work directly. Do you think it is enough to leave it like that, assuming reader will understand that he has to pull the helper from the container and use it like this?